main.cpp

```diff
@@ -1,25 +1,284 @@
 #include <iostream>
+#include <cstdlib>     /* srand, rand */
+#include <ctime>       /* time */
+#include <dirent.h>
+#include <fstream>
+
+// code from https://gist.github.com/Nircek
+// under MIT license
+#include <iostream>
+#ifdef _WIN32
+#include <conio.h>
+#else
+#include <unistd.h>
+#include <termios.h>
+char getch(){
+    // src: https://stackoverflow.com/a/16361724/6732111
+    char buf=0;
+    struct termios old={0};
+    std::cout.flush();
+    if(tcgetattr(0, &old)<0)
+        throw "tcsetattr()";
+    old.c_lflag&=~ICANON;
+    old.c_lflag&=~ECHO;
+    old.c_cc[VMIN]=1;
+    old.c_cc[VTIME]=0;
+    if(tcsetattr(0, TCSANOW, &old)<0)
+        throw "tcsetattr ICANON";
+    if(read(0,&buf,1)<0)
+        throw "read()";
+    old.c_lflag|=ICANON;
+    old.c_lflag|=ECHO;
+    if(tcsetattr(0, TCSADRAIN, &old)<0)
+        throw "tcsetattr ~ICANON";
+    std::cout<<buf;
+    return buf;
+ }
+#endif
 
 using namespace std;
 
+//struktury
+typedef struct{int x; int y;} COORDS;
+
 // zmienne globalne
-string mapa[100];
-int coordx, coordy;
-int endcoordx, endcoordy;
-char user='X';
-string file_name="plik";
+string mapy="maps"; //folder, w którym są mapy
+string file_name;   //ścieżka do mapy
+string prev_file;   //nazwa poprzedniej mapy
+#define SIZE 100
+string mapa[SIZE], bufor[SIZE];
+char user_ch='X', end_ch='O';
+COORDS player_coords;
+COORDS end_coords;
 
 // procedury
-void getMap(); // wczytuje mapę z pliku file_name i zapisuje do mapa[]
-void beginingcoords(); // bierze poczontkowew koordy i wrzuca do zmiennej
-void targetcoords(); // bierze koordy celu/wyjścia i wrzuca do zmiennej
-bool isWall(int x, int y);  // czy na podanych kordach nie ma spacji
-char drawOnBufor(int x, int y, char c); // narysuj na x i y znak c i zwróć poprzednie co tam było
-void viewBufor(); // wypisz bufor na ekran
-char getEvent(); // pobierz znak
-void doEvent(); // wykonaj operację przypisaną do danego znaku (np WSAD)
+void getMap()
+{
+std::fstream file;
+file.open(file_name, ios::in);
```

Comment:
0.28s$ ./make o
main.cpp: In function ‘void getMap()’:
main.cpp:58:29: error: no matching function for call to ‘std::basic_fstream<char>::open(std::string&, const openmode&)’
 file.open(file_name, ios::in);
                             ^
main.cpp:58:29: note: candidate is:
In file included from main.cpp:5:0:
/usr/include/c++/4.8/fstream:886:7: note: void std::basic_fstream<_CharT, _Traits>::open(const char*, std::ios_base::openmode) [with _CharT = char; _Traits = std::char_traits<char>; std::ios_base::openmode = std::_Ios_Openmode]
       open(const char* __s
       ^
/usr/include/c++/4.8/fstream:886:7: note:   no known conversion for argument 1 from ‘std::string {aka std::basic_string<char>}’ to ‘const char*’
GET 1
The command "./make o" exited with 1.

Reply:
@MrDarkness19 mówiłem Ci już kiedyś, żebyś to scastował